sktime/forecasting/compose/_reduce.py

```diff
@@ -1903,6 +1910,7 @@ def __init__(
         X_treatment="concurrent",
         impute_method="bfill",
         pooling="local",
+        windows_identical=True,
```

Comment:
one would expect that default value for bool is False :-) not against taht just to understand why it is true

Reply:
I do think that `False` (=non-identical windows) would be the more sensible default, but @fkiraly suggested in #6545 that the current behavior should remain the default.

Reply:
yes, I suppose the more sensible default indeed is `False`, but current behaviour needs to remian default due to not breaking (or changing logic) behind downstream code, see change guide:
https://www.sktime.net/en/latest/developer_guide/deprecation.html#deprecation-policy

We could think about changing it though, but then we have to go through deprecation-change handling.

Reply:
If necessary, we can go through deprecation-change handling. The current class does explicitly warn about interface changes though (`DirectReductionForecaster is experimental, and interfaces may change.`, ll. 1927 ff.).

Reply:
I see - we should still try to be nice and raise a warning that is concrete about the change. If we want to change this. What is your opinion? The `windows_identical=True` option is probably going to be more performant.

Btw, do you know whether the general performance issues have improved through other updates, e.g., boilerplate speedups?

Reply:
I've made `windows_identical=False` the default and added a warning about the change. Maybe with certain series and short horizons `windows_identical=True` may make sense, but I don't think it is a sensible default. Typically the most recent past carries important information that you'd want to use in forecasting.

Regarding the warnings in general, tbh I'd favor getting rid of the warnings altogether, the current warning is already very annoying and not easily silenced.

Regarding general performance improvements, I can't say. It's always been performant enough for our use cases. I can check but I don't think we noticed any major changes in performance over . (And compared to the old reduction classes it is impossible to say, because a like-for-like comparison is not possible due to bugs and different behavior.)

Reply:
As noted in the description, another default that is not as expected in my opinion is the fact that missings due to lagging by `window_length` are always imputed. I think there should at least be an option to drop them.

Reply:
I see: how about we remove the "experimental" warning?

Regarding the second warning:

* we need to warn in *advance*, not *after the change*. For that, let's schedule the change for 0.32.0, using the "change default" pattern: https://www.sktime.net/en/stable/developer_guide/deprecation.html#id1
* for warning, we should use `warn` from `sktime.utils.warnings` and make reference from `self`, so users can silence it via `set_config`.

Reply:
> I think there should at least be an option to drop them.
> This behavior is also different in the old reduction classes and I think should be adjusted (in another PR).

Agreed - you could contribute that?

Reply:
>     * we need to warn in _advance_, not _after the change_. For that, let's schedule the change for 0.32.0, using the "change default" pattern: https://www.sktime.net/en/stable/developer_guide/deprecation.html#id1

I've changed everything accordingly. Let's leave the experimental warning for now.

> > I think there should at least be an option to drop them.
> > This behavior is also different in the old reduction classes and I think should be adjusted (in another PR).
> 
> Agreed - you could contribute that?

I can try to have a look at it some time. I would appreciate some guidance on where this would be best implemented (e.g. whether to adjust `ReductionTransformer` or modify `Imputer` directly - are there any side effects to modifying `Imputer`?)

Reply:
> I see: how about we remove the "experimental" warning? I could be convinced if you would like to add yourself to the `maintainers` tag of the class, so it has someone who looks after it...
> 
> Regarding the second warning:
> 
>     * we need to warn in _advance_, not _after the change_. For that, let's schedule the change for 0.32.0, using the "change default" pattern: https://www.sktime.net/en/stable/developer_guide/deprecation.html#id1
> 
>     * for warning, we should use `warn` from `sktime.utils.warnings` and make reference from `self`, so users can silence it via `set_config`.

I removed the warning. Since it is thrown in the constructor, it can't be silenced via `set_config`.